src/integration/cloneDocuSources.js

```diff
@@ -70,16 +72,16 @@ module.exports = cloneDocuSources;
  * @param {Object} [config] - basic integration configuration
  * @param {Function} [cb] - callback for asynchronous operation
  */
-function _createMatrixWithNotClonedRepositories(topicDetails, config, cb) {
+function _createMatrixWithRepositories(topicDetails, tempLocation, fileName, cb) {
 
   //push to array all not cloned repositories
-  notClonedArray.push(topicDetails.clonedGenRNDestLocation, topicDetails.clonedGenRNDestLocationInternal, topicDetails.clonedGenDestLocation, topicDetails.clonedGenDestLocationInternal);
+  repositoriesArray.push(topicDetails.clonedGenRNDestLocation, topicDetails.clonedGenRNDestLocationInternal, topicDetails.clonedGenDestLocation, topicDetails.clonedGenDestLocationInternal);
 
   //push to array all not cloned repositories - latest versions (for services)
-  if (topicDetails.isService) notClonedArray.push(topicDetails.clonedGenRNDestLocationLatest, topicDetails.clonedGenRNDestLocationInternalLatest, topicDetails.clonedGenDestLocationLatest, topicDetails.clonedGenDestLocationInternalLatest);
+  if (topicDetails.isService) repositoriesArray.push(topicDetails.clonedGenRNDestLocationLatest, topicDetails.clonedGenRNDestLocationInternalLatest, topicDetails.clonedGenDestLocationLatest, topicDetails.clonedGenDestLocationInternalLatest);
 
   //write array the file
-  creator.createFile(`${config.tempLocation}/notClonedRepositories.json`, notClonedArray, (err) => {
+  creator.createFile(`${tempLocation}/${fileName}`, repositoriesArray, (err) => {
```

Comment:
this not a cool approach to always create file even if it is not needed and overwritten with each iteration. Extract to separate function and call only once at the end of the clone operation